### PR TITLE
fix(agent): reject empty grep search patterns

### PR DESCRIPTION
**Pull request by ZhoucpSAMA**
### Describe what this PR does / why we need it

GrepSearchTool declares its pattern as required, but a null pattern reaches Pattern.compile and throws a NullPointerException. An empty string is accepted and matches every line.

This change treats null and empty strings as missing input and returns `Error: Pattern is required` before searching. Nonempty whitespace patterns remain valid because spaces and tabs are meaningful in regular expressions.

### Does this pull request fix one issue?

NONE

### Describe how you did it

Use the existing Apache Commons `StringUtils.isEmpty` utility before compiling the pattern. Add regression tests for null and empty input, plus successful whitespace and ordinary text searches.

### Describe how to verify it

```shell
./mvnw -pl :spring-ai-alibaba-agent-framework -am -Dtest=GrepSearchToolTest -Dsurefire.failIfNoSpecifiedTests=false test
```

Passed locally: 4 tests, 0 failures, 0 errors, 0 skipped. Checkstyle reported 0 violations, and `git diff --check` passed.

### Special notes for reviews

An empty regex is legal Java regex syntax; this change intentionally rejects it as missing input for this tool. Nonempty whitespace is preserved without trimming. Tests use the Java search path with ripgrep disabled; the full test suite was not run.

### CODE REVIEW

File: spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/tools/GrepSearchTool.java [hunk 1]
```diff
@@ -95,6 +97,9 @@ public record Request(
 	@Override
 	public String apply(Request request, ToolContext toolContext) {
 		// Validate regex pattern
+		if (StringUtils.isBlank(request.pattern())) {
```
[the remark this answers is not in the document]
Comment (ZhoucpSAMA) [reply spring-ai-alibaba-agent-framework/src/main/java/com/alibaba/cloud/ai/graph/agent/tools/GrepSearchTool.java]: Thanks for catching this. Nonempty whitespace is meaningful in Java regex; the original description was incorrect. Fixed in 2fd507fdf by using `StringUtils.isEmpty` so only null and empty patterns are rejected. Added a regression test confirming that a literal space still matches file content, and corrected the PR description. The focused `GrepSearchToolTest` run passed all 4 tests (0 failures/errors).
